### ISSUE TITLE
deprovisioning router resources doesn't work in standard-space

### ISSUE THREAD

**Issue by tkratky** [EnMasseProject/enmasse#1014]
reproducible via automated test in **scale-router-test** branch
`make MULTITENANT=true SYSTEMTEST_ARGS='standard.AnycastTest#testScaleRouterAutomatically' systemtests`

what the test does:
1. in standard-space deploy 210 addresses(type anycast) //router credit = 0.01 => 210 * 0.01 = 2.1 pods - router is scaled to 3 pods as expected
2. remove all addresses 
 OR
3. remove 111 addresses //router credit =>2.1-1.11 => 0.99 pods - **router is still scaled to 3 pods**

in log of standard-controller is following exception (which is probably unrelated to deprovision issue)
```
2018-03-07 12:17:24 INFO  AddressController:82 - Check addresses in address space controller: [small-anycast-100, small-anycast-101, small-anycast-102, small-anycast-103, small-anycast-104, small-anycast-105, small-anycast-106, small-anycast-107, small-anycast-108, small-anycast-109, small-anycast-110, small-anycast-111, small-anycast-112, small-anycast-113, small-anycast-114, small-anycast-115, small-anycast-116, small-anycast-117, small-anycast-118, small-anycast-119, small-anycast-120, small-anycast-121, small-anycast-122, small-anycast-123, small-anycast-124, small-anycast-125, small-anycast-126, small-anycast-127, small-anycast-128, small-anycast-129, small-anycast-130, small-anycast-131, small-anycast-132, small-anycast-133, small-anycast-134, small-anycast-135, small-anycast-136, small-anycast-137, small-anycast-138, small-anycast-139, small-anycast-200, small-anycast-201, small-anycast-202, small-anycast-203, small-anycast-204, small-anycast-205, small-anycast-206, small-anycast-207, small-anycast-208, small-anycast-209, small-anycast-50, small-anycast-51, small-anycast-52, small-anycast-53, small-anycast-54, small-anycast-55, small-anycast-56, small-anycast-57, small-anycast-58, small-anycast-59, small-anycast-61, small-anycast-62, small-anycast-63, small-anycast-64, small-anycast-65, small-anycast-70, small-anycast-71, small-anycast-72, small-anycast-73, small-anycast-74, small-anycast-75, small-anycast-76, small-anycast-77, small-anycast-78, small-anycast-79, small-anycast-80, small-anycast-81, small-anycast-82, small-anycast-83, small-anycast-84, small-anycast-85, small-anycast-86, small-anycast-87, small-anycast-88, small-anycast-89, small-anycast-90, small-anycast-91, small-anycast-92, small-anycast-93, small-anycast-94, small-anycast-95, small-anycast-96, small-anycast-97, small-anycast-98, small-anycast-99]
2018-03-07 12:17:27 WARN  ResourceController:77 - Exception doing resource update
io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: PUT at: https://kubernetes.default.svc/api/v1/namespaces/standard-shared-0/configmaps/small-anycast-134 . Message: configmaps "small-anycast-134" not found. Received status: Status(apiVersion=v1, code=404, details=StatusDetails(causes=[], group=null, kind=configmaps, name=small-anycast-134, retryAfterSeconds=null, uid=null, additionalProperties={}), kind=Status, message=configmaps "small-anycast-134" not found, metadata=ListMeta(resourceVersion=null, selfLink=null, additionalProperties={}), reason=NotFound, status=Failure, additionalProperties={}).
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.requestFailure(OperationSupport.java:470)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.assertResponseCode(OperationSupport.java:409)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:379)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:343)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleReplace(OperationSupport.java:260)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleReplace(OperationSupport.java:242)
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.handleReplace(BaseOperation.java:778)
	at io.fabric8.kubernetes.client.dsl.base.HasMetadataOperation$2.apply(HasMetadataOperation.java:109)
	at io.fabric8.kubernetes.client.dsl.base.HasMetadataOperation$2.apply(HasMetadataOperation.java:104)
	at io.fabric8.kubernetes.api.model.DoneableConfigMap.done(DoneableConfigMap.java:26)
	at io.fabric8.kubernetes.api.model.DoneableConfigMap.done(DoneableConfigMap.java:5)
	at io.fabric8.kubernetes.client.dsl.base.HasMetadataOperation.replace(HasMetadataOperation.java:116)
	at io.fabric8.kubernetes.client.dsl.base.HasMetadataOperation.replace(HasMetadataOperation.java:37)
	at io.enmasse.k8s.api.ConfigMapAddressApi.replaceAddress(ConfigMapAddressApi.java:107)
	at io.enmasse.controller.standard.AddressController.resourcesUpdated(AddressController.java:109)
	at io.enmasse.k8s.api.ResourceController.run(ResourceController.java:72)
	at java.lang.Thread.run(Thread.java:748)
2018-03-07 12:17:27 INFO  AddressController:82 - Check addresses in address space controller: [small-anycast-120, small-anycast-121, small-anycast-122, small-anycast-123, small-anycast-124, small-anycast-125, small-anycast-126, small-anycast-127, small-anycast-128, small-anycast-129]
2018-03-07 12:17:28 INFO AddressController:82 - Check addresses in address space controller: []
```

**Comment by tkratky** [comment]
@lulf this task could be closed: please add comment with link to PR/commit with fix
thank you

**Comment by lulf** [comment]
This was fixed by #1058
